### ISSUE TITLE
Suggestion/Request: Custom Template Components

### ISSUE THREAD

**Issue by 3DEsprit** [decosoftware/deco-ide#101]
Just an idea/request/suggestion: the ability to create a custom template as a component. They could be prefixed with something like "temp-", or "snip-" to keep separate from components, but allow for users to create their own Native templates for default components.

As a possibly poor example, making a component template: I'm a big fan of adding PropTypes and DefaultProps outside of the class, and could create a template that added the class-type with inline default export, and have my object.propTypes, and object.DefaultProps outside of the class, all by selecting a template in a new blank file. 

Maybe later there could be options to set custom properties to change on creation like the Atom editor package [Template Generator](https://atom.io/packages/template-generator) but simply having the ability to create basic templates on top of the components would offer a lot of awesome custom ability, without having to create a bunch of static settings (current component/dumb templates would make great placeholders/examples). 

Thank you for your time! I love this IDE, and thought this was something that would make it even better, and benefit several users, while fitting it's current abilities. 😃

**Comment by dabbott** [comment]
Good idea. This is a very natural extension of the existing feature, and would be pretty easy to add too.

Probably we could implement the simplest version by adding something to the `package.json`, similar to how custom components are loaded currently.

So if you have:

```
{
  "deco": {
    "templates": [
      {
        "name": "Awesome Component",
        "inputFields": {
          "hello": {
            "type": "String"
          }
        },
        "textUrl": "https://gist.githubusercontent.com/dabbott/62428b771b05b2f79627bd5a9d136c67/raw/521e169b585a79c16c58efd0bd06dbbf1350768e/component.js"
      }
    ]
  }
}
```

then Awesome Component is added to your [ + ] list and we download it on first use. The filename you enter after clicking [ + ] would be a default "input field" called `filename`, but you could potentially specify additional input fields. And then we replace `{{{filename}}}` and `{{{hello}}}` in the template code when its used. (and you can also refer to a local path, if you don't want to use a github gist to host the template)

And then step 2 would be: we use the new Deco web service (a work in progress) that saves these automatically for you so you can use them in all your projects, share them with your team, browse public ones, etc.

**Comment by 3DEsprit** [comment]
Sounds great! So much awesome! 😃 

Great to hear this fits in nicely, and can't wait to see such added, especially the Deco service. 😉

**Comment by tmaly1980** [comment]
What's the status on this? I'd love to create my own components (and especially import Native Base's components) to speed up application development. I'd much rather have control over the components than rely upon an online gallery/store to curate. Can I create a clone of the components repository and change a configuration setting somewhere in the IDE to point to my own private repo version? I'm just using it internally for my dev team so I don't see the issue with having my own repo. Ideally I wouldn't need a repo at all and it could pull the components locally as I develop as well for rapid prototyping.

Is there ANY manual way of doing this right now?
